### ISSUE TITLE
Pytorch -> ONNX -> kmodel Error, Import graph segmentation fault

### ISSUE THREAD

**Issue by yub1ng** [kendryte/nncase#154]
## Platform: 
- MacOS 10.15.5
- Pytorch==1.5.0
- ONNX Opset 9, 10, 11 all tried

## Commend:
``` sh
ncc compile weights/yolo_mini.onnx yolo_mini.kmodel -i onnx -o kmodel
```

## Error:
```
1. Import graph...
[1]    1536 segmentation fault  ncc compile weights/yolo_mini.onnx yolo_mini.kmodel -i onnx -o kmodel
```

## Model Structure
![image](https://user-images.githubusercontent.com/34766340/89122110-4ce04c00-d4f7-11ea-8d3f-9b1bf595aaed.png)

## ONNX transform process :
```
graph(%input : Float(1, 3, 240, 320),
      %backbone.0.net.0.weight : Float(16, 3, 3, 3),
      %backbone.0.net.0.bias : Float(16),
      %backbone.1.net.1.net.0.weight : Float(32, 16, 3, 3),
      %backbone.1.net.1.net.0.bias : Float(32),
      %backbone.2.net.1.net.0.weight : Float(64, 32, 3, 3),
      %backbone.2.net.1.net.0.bias : Float(64),
      %backbone.3.net.1.net.0.weight : Float(128, 64, 3, 3),
      %backbone.3.net.1.net.0.bias : Float(128),
      %backbone.4.net.1.net.0.weight : Float(256, 128, 3, 3),
      %backbone.4.net.1.net.0.bias : Float(256),
      %output.0.net.0.weight : Float(21, 256, 1, 1),
      %output.0.net.0.bias : Float(21)):
  %13 : Float(1, 16, 240, 320) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[1, 1, 1, 1], strides=[1, 1]](%input, %backbone.0.net.0.weight, %backbone.0.net.0.bias), scope: YOLOMini/Sequential[backbone]/DBL[0]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %14 : Float(1, 16, 240, 320) = onnx::LeakyRelu[alpha=0.1](%13), scope: YOLOMini/Sequential[backbone]/DBL[0]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  %15 : Float(1, 16, 120, 160) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%14), scope: YOLOMini/Sequential[backbone]/MDBL[1]/Sequential[net]/MaxPool2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:488:0
  %16 : Float(1, 32, 120, 160) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[1, 1, 1, 1], strides=[1, 1]](%15, %backbone.1.net.1.net.0.weight, %backbone.1.net.1.net.0.bias), scope: YOLOMini/Sequential[backbone]/MDBL[1]/Sequential[net]/DBL[1]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %17 : Float(1, 32, 120, 160) = onnx::LeakyRelu[alpha=0.1](%16), scope: YOLOMini/Sequential[backbone]/MDBL[1]/Sequential[net]/DBL[1]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  %18 : Float(1, 32, 60, 80) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%17), scope: YOLOMini/Sequential[backbone]/MDBL[2]/Sequential[net]/MaxPool2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:488:0
  %19 : Float(1, 64, 60, 80) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[1, 1, 1, 1], strides=[1, 1]](%18, %backbone.2.net.1.net.0.weight, %backbone.2.net.1.net.0.bias), scope: YOLOMini/Sequential[backbone]/MDBL[2]/Sequential[net]/DBL[1]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %20 : Float(1, 64, 60, 80) = onnx::LeakyRelu[alpha=0.1](%19), scope: YOLOMini/Sequential[backbone]/MDBL[2]/Sequential[net]/DBL[1]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  %21 : Float(1, 64, 30, 40) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%20), scope: YOLOMini/Sequential[backbone]/MDBL[3]/Sequential[net]/MaxPool2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:488:0
  %22 : Float(1, 128, 30, 40) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[1, 1, 1, 1], strides=[1, 1]](%21, %backbone.3.net.1.net.0.weight, %backbone.3.net.1.net.0.bias), scope: YOLOMini/Sequential[backbone]/MDBL[3]/Sequential[net]/DBL[1]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %23 : Float(1, 128, 30, 40) = onnx::LeakyRelu[alpha=0.1](%22), scope: YOLOMini/Sequential[backbone]/MDBL[3]/Sequential[net]/DBL[1]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  %24 : Float(1, 128, 15, 20) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%23), scope: YOLOMini/Sequential[backbone]/MDBL[4]/Sequential[net]/MaxPool2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:488:0
  %25 : Float(1, 256, 15, 20) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[3, 3], pads=[1, 1, 1, 1], strides=[1, 1]](%24, %backbone.4.net.1.net.0.weight, %backbone.4.net.1.net.0.bias), scope: YOLOMini/Sequential[backbone]/MDBL[4]/Sequential[net]/DBL[1]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %26 : Float(1, 256, 15, 20) = onnx::LeakyRelu[alpha=0.1](%25), scope: YOLOMini/Sequential[backbone]/MDBL[4]/Sequential[net]/DBL[1]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  %27 : Float(1, 21, 15, 20) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[1, 1], pads=[0, 0, 0, 0], strides=[1, 1]](%26, %output.0.net.0.weight, %output.0.net.0.bias), scope: YOLOMini/Sequential[output]/DBL[0]/Sequential[net]/Conv2d[0] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/modules/conv.py:342:0
  %out : Float(1, 21, 15, 20) = onnx::LeakyRelu[alpha=0.1](%27), scope: YOLOMini/Sequential[output]/DBL[0]/Sequential[net]/LeakyReLU[1] # /Users/qiyubing/opt/anaconda3/envs/deep_learning/lib/python3.7/site-packages/torch/nn/functional.py:1063:0
  return (%out)
```

**Comment by sunnycase** [comment]
Can you upload your model?

**Comment by yub1ng** [comment]
> Can you upload your model?

Sure, Thanks for your reply

[yolo_mini.py.zip](https://github.com/kendryte/nncase/files/5014088/yolo_mini.py.zip)

**Comment by yub1ng** [comment]
After compiled from master, the error change to 
```
1. Import graph...
Fatal: Shapes must be same, but got [] and [1x16x240x320]
```
That's might be a good sign that the last problem has been solved.
But the real shape of model output is 
```%main_output : Float(1, 21, 15, 20) = onnx::LeakyRelu[alpha=0.10000000000000001](%27)
```
How can I solve this?

**Comment by yub1ng** [comment]
After compiled from master, the error change to 
```
1. Import graph...
Fatal: Shapes must be same, but got [] and [1x21x15x20]
```
That's might be a good sign that the last problem has been solved.
But How can I solve this?

**Comment by sunnycase** [comment]
Fixed in 335b80ddcdd6ebe64c9ea2e581c317f45ff60b8a.

**Comment by yub1ng** [comment]
After my test, if I use `nn.LeakyReLU()`, there will be a shape error. the error will disappear when I use `nn.RELU()` instead of `LeakyReLU`

Reproduce Example:
~~~python
nn.Sequential(
            nn.Conv2d(3, 16, kernel_size=3, stride=1, padding=1),
            nn.LeakyReLU(0.1),
            nn.Conv2d(16, 8, kernel_size=3, stride=1, padding=1),
        )
~~~
It will cause `Fatal: Shapes must be same, but got [] and [1x16x240x320]`

~~~python
nn.Sequential(
            nn.Conv2d(3, 16, kernel_size=3, stride=1, padding=1),
            nn.ReLU(),
            nn.Conv2d(16, 8, kernel_size=3, stride=1, padding=1),
        )
~~~
This will succeed.

**Comment by yub1ng** [comment]
> Fixed in [335b80d](https://github.com/kendryte/nncase/commit/335b80ddcdd6ebe64c9ea2e581c317f45ff60b8a).

Ha, thanks! The LeakyReLU Error has been solved, but the SUMMARY `ncc` printed just show the input and output, does it work correctly? I saw others will see the entire model structure including hidden layer information of their model.

Here is I've got:
```bash
SUMMARY
INPUTS
0	Input_0	1x3x240x320
OUTPUTS
0	main_output	1x21x15x20
```

**Comment by sunnycase** [comment]
> Ha, thanks! The LeakyReLU Error has been solved, but the SUMMARY `ncc` printed only shows the input and output, does it work correctly? I saw others will see the entire model structure including hidden layer information of their model.

It's by design. v0.2 doesn't print the whole structure.

**Comment by yub1ng** [comment]
> > Ha, thanks! The LeakyReLU Error has been solved, but the SUMMARY `ncc` printed only shows the input and output, does it work correctly? I saw others will see the entire model structure including hidden layer information of their model.
> 
> It's by design. v0.2 doesn't print the whole structure.

THX!!!!
